Doc updates 1

akeeste opened the pull request:
This PR
- [x] resolves #54
- [x] resolves #55 
- [x] resolves #59
- [x] updates the required Python version
- [x] Ensure all external links work
- [x] Ensure all internal links work
- [ ] rebuild documentation

simmsa commented:
@akeeste, sorry for not pointing this out earlier. The fix is here: https://github.com/MHKiT-Software/MHKiT/pull/58/commits/1896ee1963185f653e79325d651133ed9c224b2f. Annoyingly it seems like some sphinx function is case sensitive.

akeeste commented:
@simmsa Thanks! I don't see that commit on the master branch but added it to this PR. 

The mooring module is now showing up, but has some incorrect formatting within the example so it shows up on the table of contents incorrectly.

akeeste commented:
@simmsa can you give this a review? I have not rebuilt the docs yet so that the changes are easier to review. If things look good, I'll rebuild, double check for issues, then move on to some reorganization tasks I want to address.

simmsa commented on docs/source/installation.rst:
```diff
@@ -16,22 +20,23 @@ Refer to the `Python website <https://www.python.org/>`_ for information on usin
 
 .. Note::
```
Does this note still add value?

simmsa commented:
@akeeste, this looks good to me, nice work! My only question is related to the second note within the python installation instructions. Is this note still necessary (screenshot below)? MacOS [Removed Python 2.7 Support in Monterey 12.3 on 3/14/2023](https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes#Python) so maybe now is a good time to remove this section? To me this feels outdated.
![Screenshot 2024-05-14 at 4 50 04 PM](https://github.com/MHKiT-Software/MHKiT/assets/3704572/d6e8d9a4-f128-44b7-9cfb-17e5eea72143)

simmsa commented:
@akeeste, I agree with removing the windows path section. I pulled this branch locally verified that everything looks correct. I don't see any formatting or build errors. Nice work!

akeeste commented:
FYI I tried to do something clever here with submodules, which make the automated build/deployment fail. I reverted the python submodule to master so that the updated docs deploy. But the mooring example will not show up accurately.

simmsa commented:
Good fix! For me the mooring example looks and performs nominally in the deployed docs on Chrome. The rendering method of the animations does seems fragile, but it looks like it works.

akeeste commented:
Noting what I did in this PR and then [here](https://github.com/MHKiT-Software/MHKiT/commit/3154dacdba99ab654b46c2e7227980b97f5ad641) because the automated build/deploy is not performing as I would expect. Something weird is happening.

In this PR I:
- Updated the submodules to pull from the master branch of each MHKiT-Python and MHKiT-MATLAB as normal
- Manually went into the Python submodule and added a commit that fixed the formatting in mooring_example.ipynb
   -   The top level title (`#`) was incorrectly added for the section title "Calculating the Lay Length with MoorDyn Data". This made the title appear in the Examples TOC as another example when it should not have appeared.
- Committed the change of the MHKiT-Python submodule at the MHKiT repo level
- Rebuilt the docs. The Mooring example formatting was fixed.
- Committed all of the built docs.
- Pushed to this PR and merged.

Then:
- Actions [would not deploy the docs](https://github.com/MHKiT-Software/MHKiT/actions/runs/9098213594) because the automated build failed when the submodule had a different commit number than expected (it expected the commit number on MHKiT-Python master branch).
- I pushed a commit to MHKiT master that reverted the submodule to the commit currently on MHKiT-Python master. I did not rebuild the docs manually.
- The docs automatically [rebuilt and deployed](https://github.com/MHKiT-Software/MHKiT/actions/runs/9098585816). But the mooring example has _not_ reverted to the format expected given the notebook in the Python submodule

So, there seems to be an issue with the automated building. The deployed docs are not those being automatically built, but those being manually built and committed. This would also explain why #54 had to be opened even though the mooring example was called out in [python.rst in v0.7.0](https://github.com/MHKiT-Software/MHKiT/blob/aa0ca260051156df270bc62dd969490b814cac77/docs/source/python.rst?plain=1)
